lib/cartopy/crs.py

```diff
@@ -459,6 +459,11 @@ def _rings_to_multi_polygon(self, rings, is_ccw):
                 if prep_polygon.contains(interior_ring):
                     holes.append(interior_ring)
                     interior_rings.remove(interior_ring)
+                elif polygon.crosses(interior_ring):
```

Comment:
Sounds sensible. I don't know how much of an overhead prepping is, but logic suggests that there shouldn't be too much difference between ``prep(poly.ext).intersects`` and ``poly.crosses``.